src/main/java/io/nats/service/ServiceResponse.java

```diff
@@ -59,6 +63,16 @@ protected ServiceResponse(String type, JsonValue jv) {
         name = Validator.required(readString(jv, NAME), "Name");
         version = Validator.required(readString(jv, VERSION), "Version");
         metadata = readStringStringMap(jv, METADATA);
+        serialized = new AtomicReference<>();
+    }
+
+    @Override
+    public byte[] serialize() {
+        // lazy since endpoints can be added after creation
+        if (serialized.get() == null) {
+            serialized.set(toJson().getBytes(StandardCharsets.UTF_8));
+        }
+        return serialized.get();
```

Comment:
This method is part of the JsonSerializable interface, which has a default, always execute implementation. This is why I originally applied the serializer in the add discovery, so it didn't need to be executed every time. But since we can add endpoints, the serialization needs to be updated when endpoints are added.